packages/core/src/__integrationtests__/Ctypes.spec.ts

```diff
@@ -23,8 +23,17 @@ import { disconnect } from '../kilt'
 import { createEndowedTestAccount, initializeApi, submitTx } from './utils'
 
 let api: ApiPromise
+// we skip fetching CTypes from chain for the old pallet version, where the necessary information was not yet on chain.
+let hasBlockNumbers: boolean
 beforeAll(async () => {
   api = await initializeApi()
+  // this extrinsic should only exist in the new pallet version
+  hasBlockNumbers = typeof api.tx.ctype.setBlockNumber !== 'undefined'
+  if (!hasBlockNumbers) {
+    console.warn(
+      'detected pallet version 1, skipping CType fetching which is not yet supported'
```

Comment:
Version could be either 0 or 1 here, since 2 is the first valid one containing block number.

Reply:
but 0 is an invalid version number technically, if I got that right?